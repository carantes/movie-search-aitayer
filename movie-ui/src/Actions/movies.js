import { REQUEST_MOVIES, RECEIVE_MOVIES, RECEIVE_ERROR } from "./types";
import { fetchHelper } from "Utils";

function requestMovies(keyword) {
  return {
    type: REQUEST_MOVIES,
    payload: keyword,
    loading: true,
  };
}

function receiveMovies(movies, keyword) {
  return {
    type: RECEIVE_MOVIES,
    movies,
    keyword: keyword,
    loading: false,
  };
}

function receiveError(err) {
  return {
    type: RECEIVE_ERROR,
    error: err,
    loading: false,
  };
}

function parseData(movies) {
  return movies.filter(result => result !== null);
}

export const fetchGetMovies = keyword => {
  const apiURL = keyword ? `api/search?keyword=${keyword}` : `api/search`;

  return async dispatch => {
    dispatch(requestMovies(keyword));
    return await fetchHelper(apiURL)
      .then(parseData)
      .then(movies => dispatch(receiveMovies(movies, keyword)))
      .catch(err => dispatch(receiveError(err)));
  };
};
