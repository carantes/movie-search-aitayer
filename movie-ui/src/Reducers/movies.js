import {
  REQUEST_MOVIES,
  RECEIVE_MOVIES,
  RECEIVE_ERROR,
} from "../Actions/types";

export const initialState = {
  loading: false,
  movies: [],
  error: null,
};

function movies(state = initialState, action) {
  const { type, payload, loading, movies, error } = action;
  switch (type) {
    case REQUEST_MOVIES:
      return {
        payload,
        loading,
      };
    case RECEIVE_MOVIES:
      return {
        payload,
        movies,
        loading,
      };
    case RECEIVE_ERROR:
      return {
        error,
        loading,
      };
    default:
      break;
  }
  return state;
}

export default movies;
