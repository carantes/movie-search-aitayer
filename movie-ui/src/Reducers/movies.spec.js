import movies from "./movies";
import { initialState } from "./movies";
import {
  REQUEST_MOVIES,
  RECEIVE_MOVIES,
  RECEIVE_ERROR,
} from "../Actions/types";

describe("Movies reducer", () => {
  it("should return the initial state", () => {
    expect(movies(undefined, {})).toMatchSnapshot();
  });

  it("should handle REQUEST_MOVIES", () => {
    expect(
      movies(initialState, {
        type: REQUEST_MOVIES,
      }),
    ).toMatchSnapshot();
  });

  it("should handle RECEIVE_MOVIES", () => {
    expect(
      movies(initialState, {
        type: RECEIVE_MOVIES,
      }),
    ).toMatchSnapshot();
  });

  it("should handle RECEIVE_ERROR", () => {
    expect(
      movies(initialState, {
        type: RECEIVE_ERROR,
      }),
    ).toMatchSnapshot();
  });
});
