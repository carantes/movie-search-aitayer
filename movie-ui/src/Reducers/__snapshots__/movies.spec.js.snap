// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies reducer should handle RECEIVE_ERROR 1`] = `
Object {
  "error": undefined,
  "loading": undefined,
}
`;

exports[`Movies reducer should handle RECEIVE_MOVIES 1`] = `
Object {
  "loading": undefined,
  "movies": undefined,
  "payload": undefined,
}
`;

exports[`Movies reducer should handle REQUEST_MOVIES 1`] = `
Object {
  "loading": undefined,
  "payload": undefined,
}
`;

exports[`Movies reducer should return the initial state 1`] = `
Object {
  "error": null,
  "loading": false,
  "movies": Array [],
}
`;
